### PR TITLE
fix(core-transaction-pool-mem): sort transactions by numerical fee value

### PR DESCRIPTION
**Pull request by zillionn**
In [getTransactionsOrderedByFee()](https://github.com/ArkEcosystem/core/blob/0ab42975b23a479c6168fb5a73ebbbc51932f031/packages/core-transaction-pool-mem/lib/mem.js#L218) `a.transaction.fee` and `b.transaction.fee` are BigNumber objects not numbers and the comparison fails.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist

- [ ] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/core-transaction-pool-mem/lib/mem.js [hunk 1]
```diff
@@ -218,10 +219,10 @@ class Mem {
   getTransactionsOrderedByFee () {
     if (!this.allIsSorted) {
       this.all.sort(function (a, b) {
-        if (a.transaction.fee > b.transaction.fee) {
+        if (+bignumify(a.transaction.fee).toFixed() > +bignumify(b.transaction.fee).toFixed()) {
           return -1
         }
-        if (a.transaction.fee < b.transaction.fee) {
+        if (+bignumify(a.transaction.fee).toFixed() < +bignumify(b.transaction.fee).toFixed()) {
```
Comment (faustbrian) [inline packages/core-transaction-pool-mem/lib/mem.js]: http://mikemcl.github.io/bignumber.js/#lt

File: packages/core-transaction-pool-mem/lib/mem.js [hunk 2]
```diff
@@ -218,10 +219,10 @@ class Mem {
   getTransactionsOrderedByFee () {
     if (!this.allIsSorted) {
       this.all.sort(function (a, b) {
-        if (a.transaction.fee > b.transaction.fee) {
+        if (+bignumify(a.transaction.fee).toFixed() > +bignumify(b.transaction.fee).toFixed()) {
```
Comment (faustbrian) [inline packages/core-transaction-pool-mem/lib/mem.js]: http://mikemcl.github.io/bignumber.js/#gt

**Comment by faustbrian** [comment]
I resolved the test issues again this time. Next time run the tests locally before you send a PR and make sure they pass.

**Comment by zillionn** [comment]
@faustbrian Sorry, I just saw the bug but it was 2:20 AM here...
@boldninja Don't track my solution for the bounty.

**Pull request merged**